### PR TITLE
misc dependency changes

### PR DESCRIPTION
**Pull request by bryevdv**
Some small changes and clarifications.

### CODE REVIEW

**Comment by bryevdv** [comment]
1.9.3 is the min numpy version in defaults for python 3.6, we could consider bumping the min requirement to that as well.

File: scripts/ci/install.test [hunk 1]
```diff
@@ -35,7 +35,11 @@ else
 fi
 
 # install the pre-built Bokeh package plus all test and runtime dependencies
-conda install -n testenv --yes --quiet --use-local bokeh `python scripts/deps.py run test` tornado=${TORNADO:-5}
+if [[ -z "${NUMPY}" ]]; then
+    conda install -n testenv --yes --quiet --use-local bokeh `python scripts/deps.py run test` tornado=${TORNADO:-6}
+else
+    conda install -n testenv --yes --quiet --use-local bokeh `python scripts/deps.py run test` tornado=${TORNADO:-6} numpy=${NUMPY}
```
Comment (mattpap) [inline scripts/ci/install.test]: This could be coalesced to avoid repetition:
```sh
 conda install (...) tornado=${TORNADO:-6} numpy=${NUMPY:-*}
```

Comment (bryevdv) [reply scripts/ci/install.test]: will change

**Comment by bryevdv** [comment]
There is actually a few tests failing with 1.9.3
```
____________ TestColumnDataSource.test__stream_good_datetime64_data ____________
self = <test_sources.TestColumnDataSource object at 0x7f824bdb5080>
    def test__stream_good_datetime64_data(self):
        now = dt.datetime.now()
>       dates = np.array([now+dt.timedelta(i) for i in range(1, 10)], dtype='datetime64')
E       TypeError: Cannot cast datetime.datetime object from metadata [us] to [D] according to the rule 'same_kind'
unit/bokeh/models/test_sources.py:359: TypeError
______ TestColumnDataSource.test__stream_good_datetime64_data_transformed ______
self = <test_sources.TestColumnDataSource object at 0x7f824bd1f278>
    def test__stream_good_datetime64_data_transformed(self):
        now = dt.datetime.now()
>       dates = np.array([now+dt.timedelta(i) for i in range(1, 10)], dtype='datetime64')
E       TypeError: Cannot cast datetime.datetime object from metadata [us] to [D] according to the rule 'same_kind'
unit/bokeh/models/test_sources.py:376: TypeError
_______________________ test_convert_datetime_type_array _______________________
    def test_convert_datetime_type_array():
>       a = np.array(['2018-01-03T15:37:59', '2018-01-03T15:37:59.922452', '2016-05-11'], dtype='datetime64')
E       TypeError: Cannot parse "2016-05-11" as unit 'us' using casting rule 'same_kind'
unit/bokeh/util/test_serialization.py:136: TypeError
```
These all actually seem to be in the formulation of tests data in the tests themselves so I will see about updating these few tests.

**Comment by bryevdv** [comment]
OK, 1.11.3 should pass with no test changes required and is still well, well older than the latest version the numpy considers supported or that is currently listed in NEP 25

**Comment by bryevdv** [comment]
OK I am going to go ahead and merge this. MPL specifies np>= 1.12 so I think this should not be too conroversial a change.

**Comment by bryevdv** [comment]
For ref here are some recent conda pkg numbers by version, I think 1.11.3 is fine

Screen-Shot-2019-11-15-at-9 59 25-AM

**Pull request merged**